lib/Server.js

```diff
@@ -69,6 +69,17 @@ class Server {
 
     normalizeOptions(this.compiler, this.options);
 
+    // don't move this position because addEntries called by updateCompiler checks this.options.hot|hotOnly
+    this.hot =
+      typeof this.options.hot === 'boolean' ||
+      typeof this.options.hotOnly === 'boolean'
+        ? this.options.hot || this.options.hotOnly
+        : true;
+
+    if (this.hot) {
+      this.options.hot = true;
+    }
```

Comment:
Why do not use `this.options.hot` only (avoid using `this.hot`)?

Reply:
addEntries checks this.options.hot so we need to add true when this.hot is true.  So you want to delete hotOnly or hotOnly should be added true as well?

Reply:
Can we rewrite code on using `this.options.hot` only. What do you think about - `this.options.hot: boolean | "only"`?

Reply:
ok, understood. so we need to rewrite this line(https://github.com/webpack/webpack-dev-server/blob/89ffb86cd26425c59e3937ca06a2c804a01b8f1d/lib/Server.js#L734). Can we delete hotOnly...?

Reply:
Ideally - avoid using `this.[propertyFromOptions]` in favor `this.options.[property]` in all code, but we can do it in the separate PR

Reply:
Yes, I'll work on this on Next branch.

Reply:
Please remove `this.hot` in thus PR in favor `this.options.hot`